core/src/main/scala/kafka/server/QuotaFactory.scala

```diff
@@ -0,0 +1,57 @@
+package kafka.server
+
+import kafka.common.TopicAndPartition
+import org.apache.kafka.common.metrics.Metrics
+import org.apache.kafka.common.utils.SystemTime
+
+import scala.collection.Map
+
+object QuotaFactory {
+
+  object QuotaType extends Enumeration {
```

Comment:
The Scala `Enumeration` class has a bunch of problems and we typically use ADTs. An example is `RackAwareMode`.